cmake/modules/FindBabbleSim.cmake

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# FindBabbleSim module for locating BabbleSim
+#
+# The module defines the following variables:
+#
+# 'BSIM_COMPONENTS_PATH'
+# Path to the BabbleSim components source folder
+#
+# 'BSIM_OUT_PATH'
+# Path to the BabbleSim build output root path (under which libraries and binaries) are kept
+#
+# We first try to find it via the environment variables BSIM_OUT_PATH and BSIM_COMPONENTS_PATH.
+# If these are not set, as a fallback we attempt to find it thru west, in case the user
+# fetched babblesim using the manifest.
+# Note that what we find thru the environment variables is meant to have precedence.
+#
+# If BabbleSim cannot be found we error right away with a message trying to guide users
+
+zephyr_get(BSIM_COMPONENTS_PATH)
+zephyr_get(BSIM_OUT_PATH)
+
+if ((DEFINED WEST) AND (NOT DEFINED BSIM_COMPONENTS_PATH) AND (NOT DEFINED BSIM_OUT_PATH))
+  # Let's ask west for the bsim_project existence and its path
+  execute_process(COMMAND ${WEST}
+    status babblesim_base
+    OUTPUT_QUIET
+    RESULT_VARIABLE ret_val1)
+  execute_process(COMMAND ${WEST}
+    list babblesim_base -f {posixpath}
+    OUTPUT_VARIABLE BSIM_BASE_PATH
+    RESULT_VARIABLE ret_val2)
```

Comment:
Note that the output from these commands is displayed to the end user, e.g.:
```
-- The CXX compiler identification is GNU 12.2.1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/gcc
FATAL ERROR: unknown project name/path: babblesim_base
  Hint: use "west list" to list all projects.
FATAL ERROR: unknown project name/path: babblesim_base
  Hint: use "west list" to list all projects.
-- Using BSIM from BSIM_COMPONENTS_PATH= BSIM_OUT_PATH=
CMake Error at /tmp/aa/zephyr/cmake/modules/FindBabbleSim.cmake:44 (message):
  This board requires the BabbleSim simulator.  Please either
```
Is this desired?